addons/website/models/website_menu.py

```diff
@@ -80,8 +81,16 @@ def unlink(self):
     @api.one
     def _compute_visible(self):
         visible = True
+
         if self.page_id and not self.page_id.sudo().is_visible and not self.user_has_groups('base.group_user'):
             visible = False
+
+        if self.group_ids:
+            user = self.env.user
+            # Loops over all groups from the current user. If any of the security groups on the menu.item record
```

Comment:
you code do exactly the opposite.

Loops over all groups from the current user.
=> we look to the list of groups on the menu record and not the current user

If any of the security groups on the menu.item record
==> if ALL of the security



ps: read my comment on pr directly ;)